cassandra/src/main/scala/akka/stream/alpakka/cassandra/GuavaFutures.scala

```diff
@@ -6,9 +6,24 @@ package akka.stream.alpakka.cassandra
 
 import com.google.common.util.concurrent.{FutureCallback, Futures, ListenableFuture}
 
-import scala.concurrent.{Future, Promise}
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success, Try}
 
 private[cassandra] object GuavaFutures {
+
+  def invokeTryCallback[T](
+      listenableFuture: ListenableFuture[T],
+      executor: java.util.concurrent.Executor = ExecutionContext.global
+  )(callback: Try[T] => Unit): Unit =
+    Futures.addCallback(
+      listenableFuture,
+      new FutureCallback[T] {
+        override def onSuccess(result: T): Unit = callback(Success(result))
+        override def onFailure(t: Throwable): Unit = callback(Failure(t))
+      },
+      executor
+    )
+
   implicit final class GuavaFutureOpts[A](val guavaFut: ListenableFuture[A]) extends AnyVal {
```

Comment:
Is this needed anymore?

Reply:
yes. it is used in Sink:
https://github.com/akka/alpakka/blob/2dac1e082b2afcb419610e8221844ae22f8338cb/cassandra/src/main/scala/akka/stream/alpakka/cassandra/scaladsl/CassandraSink.scala#L24